### PR TITLE
break modules out a bit

### PR DESCRIPTION
**Pull request by sndrs**

### CODE REVIEW

File: src/actions/validate-pull-request.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	/*************************************/
+
+	if (
+		!pullRequest.user ||
+		pullRequest.user.login !== config.pullRequestAuthor
+	) {
+		info(
+			`Pull request is not authored by ${config.pullRequestAuthor}, ignoring.`,
+		);
+		return;
```
Comment (jamie-lynch) [inline src/actions/validate-pull-request.ts:34]: Now that the validate and merge logic is split out, these returns in the validate function won't skip the merge. These are for the cases where the PR isn't a version bump PR so I don't think we should fail, just exit without doing anything else.

**Review by jamie-lynch — COMMENTED** [review]
This is great! The only other thing that I spotted is that there are a couple of test files which probably need updating to match the new structure.

File: src/actions/validate-pull-request.ts (hunk 1 above)
Comment (sndrs) [reply src/actions/validate-pull-request.ts:34]: ah yeah very good point – will look at that today

File: src/lib/pkg.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable import/order -- some rules are in conflict, but it's fine */
+import type { PackageJson } from 'type-fest';
+
+export const name =
```
Comment (jamie-lynch) [inline src/lib/pkg.ts:4]: This is nice! Would this approach work for getting the package version too?

File: src/actions/validate-pull-request.ts (hunk 1 above)
Comment (sndrs) [reply src/actions/validate-pull-request.ts]: should be resolved by https://github.com/guardian/actions-merge-release-changes-to-protected-branch/pull/30/commits/bee3fe23662df04d13dc428abc0f1839de18ae42

File: src/lib/pkg.ts (hunk 2 above)
Comment (sndrs) [reply src/lib/pkg.ts:4]: this gets _this_ package info, it's bandled by webpack. the tricky bit is getting webpack not to try to optimise away the `require` that loads package of the source this action is checking

File: src/actions/validate-pull-request.ts (hunk 1 above)
Comment (jamie-lynch) [reply src/actions/validate-pull-request.ts]: Nice 👍

File: src/lib/pkg.ts (hunk 2 above)
Comment (sndrs) [reply src/lib/pkg.ts:4]: i bet there's a npm module that would handle this though...

Comment (jamie-lynch) [reply src/lib/pkg.ts:4]: Ah yes, fair enough. I've got a separate point on the checklist to improve the way we get the version number so let's leave it for now and pick it up as it's own task later.

**Review by jamie-lynch — APPROVED** [review]
This looks great 👍

**Pull request merged**